src/types.jl

```diff
@@ -160,6 +160,23 @@ struct RGB4{T<:Fractional} <: AbstractRGB{T}
 end
 RGB4(r::T, g::T, b::T) where {T<:Fractional} = RGB4{T}(r, g, b)
 
+"""`RGB16` is Red-Green-Blue colorspace represented by two bytes (16 bits).
+
+1        |1          |        0
+5 4 3 2 1|0 9 8 7 6 5|4 3 2 1 0 
+---------|-----------|---------
+R R R R R|G G G G G G|B B B B B
+"""
+struct RGB16
+    value::UInt16
+end
+
+function RGB16(r, g, b)
+    RGB16(r << 11 + g << 5 + b)
```

Comment:
This implementation makes strong assumptions about the input types and formats. I would restrict the input arguments to `N3f5`/`N2f6` as appropriate, and then introduce "conversion" constructors for other input types.

Reply:
Also note that since

```julia
julia> typemax(N2f6)
4.05N2f6

julia> typemax(N3f5)
8.23N3f5
```

you might need to add some checks. (Both of these correspond to the bit pattern `0xff`. This is equivalent to making sure that `r`, `g`, and `b` don't have bits set that you're not expecting.)

Reply:
I won't be able to do more during the next 2 coming weeks... 
If someone here could take the lead on this that will be very nice.

Reply:
No worries, I don't think anyone else is asking for this. It can wait until you have time.